backend/app/models/memo/update_form.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class Memo
+  class UpdateForm
+    include ActiveModel::Validations
+
+    validates :memo, cascade: true
+    validates :memo_tags_to_add, cascade: true, if: -> { errors.empty? }
+
+    def initialize(params:, id:)
+      @params = params
+      @id = id
+    end
+
+    def save
+      return false if invalid?
+
+      ActiveRecord::Base.transaction do
+        save_record!(memo)
+        memo_tags_to_remove.destroy_all
+      end
+
+      errors.empty?
+    end
+
+    private
+
+    attr_reader :params, :id
+
+    def save_record!(record)
+      return true if record.save
+
+      errors.add(:base, record.error_message)
+      raise ActiveRecord::Rollback
+    end
+
+    def memo
+      @memo ||= \
+        Memo.find(id)
+            .tap do |model|
+              model.assign_attributes(
+                title: params[:title] || model.title,
+                content: params[:content] || model.content,
```

Comment:
Must

フロントからは何も編集フォームを入力しなければ、
`params[:hoge]`がnilではなく、createしたときの値のままでリクエストされるはずなので、この処理はおかしいと思います。

```suggestion
                title: params[:title],
                content: params[:content],
```

Reply:
フロントからは何も編集フォームを入力しない場合、params[:hoge]がnilになると思い込んでいました💦
修正します。